plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/TrinoDeleteFile.java

```diff
@@ -0,0 +1,317 @@
+    private final Map<Integer, Long> nanValueCounts;
+    private final Map<Integer, byte[]> lowerBounds;
+    private final Map<Integer, byte[]> upperBounds;
+    private final byte[] keyMetadata;
+    private final List<Integer> equalityFieldIds;
+    private final Integer sortOrderId;
+    private final List<Long> splitOffsets;
+
+    public static TrinoDeleteFile fromIcebergDeleteFile(DeleteFile icebergDeleteFile)
+    {
+        return new TrinoDeleteFile(
+                icebergDeleteFile.pos(),
+                icebergDeleteFile.specId(),
+                icebergDeleteFile.content(),
+                icebergDeleteFile.path(),
+                icebergDeleteFile.format(),
+                icebergDeleteFile.recordCount(),
+                icebergDeleteFile.fileSizeInBytes(),
+                icebergDeleteFile.columnSizes(),
+                icebergDeleteFile.valueCounts(),
+                icebergDeleteFile.nullValueCounts(),
+                icebergDeleteFile.nanValueCounts(),
+                icebergDeleteFile.lowerBounds() == null ? null : icebergDeleteFile.lowerBounds().entrySet().stream()
+                        .collect(toImmutableMap(Map.Entry<Integer, ByteBuffer>::getKey, entry -> entry.getValue().array())),
+                icebergDeleteFile.upperBounds() == null ? null : icebergDeleteFile.upperBounds().entrySet().stream()
+                        .collect(toImmutableMap(Map.Entry<Integer, ByteBuffer>::getKey, entry -> entry.getValue().array())),
+                icebergDeleteFile.keyMetadata() == null ? null : icebergDeleteFile.keyMetadata().array(),
+                icebergDeleteFile.equalityFieldIds(),
+                icebergDeleteFile.sortOrderId(),
+                icebergDeleteFile.splitOffsets());
+    }
+
+    @JsonCreator
+    public TrinoDeleteFile(
+            @JsonProperty("pos") Long pos,
+            @JsonProperty("specId") int specId,
+            @JsonProperty("fileContent") FileContent fileContent,
+            @JsonProperty("path") CharSequence path,
+            @JsonProperty("format") FileFormat format,
+            @JsonProperty("recordCount") long recordCount,
+            @JsonProperty("fileSizeInBytes") long fileSizeInBytes,
+            @JsonProperty("columnSizes") Map<Integer, Long> columnSizes,
+            @JsonProperty("valueCounts") Map<Integer, Long> valueCounts,
+            @JsonProperty("nullValueCounts") Map<Integer, Long> nullValueCounts,
+            @JsonProperty("nanValueCounts") Map<Integer, Long> nanValueCounts,
+            @JsonProperty("lowerBounds") Map<Integer, byte[]> lowerBounds,
+            @JsonProperty("upperBounds") Map<Integer, byte[]> upperBounds,
+            @JsonProperty("keyMetadata") byte[] keyMetadata,
+            @JsonProperty("equalityFieldIds") List<Integer> equalityFieldIds,
+            @JsonProperty("sortOrderId") Integer sortOrderId,
+            @JsonProperty("splitOffsets") List<Long> splitOffsets)
+    {
+        this.pos = pos;
```

Comment:
add `rnn` where applicable, and `@Nullable` (on constructor param and getters) elsewhere.